File: javascript_basics/src/tests/file_upload.test.js
require("../testSetup.js");
const fs = require("fs");
const path = require("path");
const { By, until } = require("selenium-webdriver");
const { getDriver } = require("../globalSetup.js");
const { logger } = require("../utils/logger.js");
const { sleep } = require("../utils/sleep.js");

describe("File Upload", function () {

  it("should upload a file successfully", async function () {

    const driver = getDriver();
    logger.info("STARTING Test --- File Upload ---");

    //
    // CREATE TEMP FILE
    //
    const tempDir = path.join(process.cwd(), "temp");
    const tempFilePath = path.join(tempDir, "upload_test_file.txt");

    logger.info(`Ensuring temp directory exists: ${tempDir}`);
    if (!fs.existsSync(tempDir)) {
      fs.mkdirSync(tempDir);
    }

    logger.info(`Creating temporary file: ${tempFilePath}`);
    fs.writeFileSync(tempFilePath, "This is a temp file for upload testing.");
    // await sleep(1000);


    //
    // OPEN PAGE
    //
    const url = "https://the-internet.herokuapp.com/upload";
    logger.info(`Navigating to ${url}`);
    await driver.get(url);
    // await sleep(1000);


    //
    // SELECT FILE + UPLOAD
    //
    logger.info("Locating file input");
    const fileInput = await driver.findElement(By.id("file-upload"));
    // await sleep(1000);

    logger.info("Sending file path to input");
    await fileInput.sendKeys(tempFilePath);
    // await sleep(1000);

    logger.info("Clicking Upload button");
    const uploadButton = await driver.findElement(By.id("file-submit"));
    await uploadButton.click();
    // await sleep(1000);


    //
    // VERIFY SUCCESS
    //
    logger.info("Waiting for success message");
    await driver.wait(until.elementLocated(By.tagName("h3")), 5000);

    const header = await driver.findElement(By.tagName("h3")).getText();
    // await sleep(1000);

    if (header.trim() !== "File Uploaded!") {
      throw new Error(`Upload failed. Expected 'File Uploaded!' but got '${header}'`);
    }

    logger.info("Upload success confirmed: File Uploaded!");


    //
    // CLEAN UP TEMP FILE
    //
    logger.info(`Deleting temporary file: ${tempFilePath}`);
    if (fs.existsSync(tempFilePath)) {
      fs.unlinkSync(tempFilePath);
    }

    logger.info("ENDING Test --- File Upload ---");
  });

});

File: csharp_basics/tests/file_upload_test.cs
using System;
using System.IO;
using NUnit.Framework;
using OpenQA.Selenium;
using csharp_basics.utils;

namespace csharp_basics.tests
{
    [TestFixture]
    public class FileUploadTest : BaseTest
    {
        [Test]
        public void TestFileUpload()
        {
            Logger.Write("STARTING Test --- File Upload ---");

            string url = "https://the-internet.herokuapp.com/upload";
            driver.Navigate().GoToUrl(url);
            Logger.Write($"Navigated to {url}");

            // Resolve project root (three levels up from bin/.../ )
            string projectRoot = AppContext.BaseDirectory;
            projectRoot = Directory.GetParent(projectRoot).Parent.Parent.Parent.FullName;

            // Create temp directory within project root
            string tempDir = Path.Combine(projectRoot, "temp");

            if (!Directory.Exists(tempDir))
            {
                Directory.CreateDirectory(tempDir);
                Logger.Write($"Created temp directory at: {tempDir}");
            }

            // Create file inside temp directory
            string tempFilePath = Path.Combine(tempDir, "sample_upload.txt");
            File.WriteAllText(tempFilePath, "This is a test file for Selenium upload automation.");
            Logger.Write($"Temporary file created at: {tempFilePath}");

            // Upload process
            IWebElement fileInput = driver.FindElement(By.Id("file-upload"));
            fileInput.SendKeys(tempFilePath);
            Logger.Write("File path sent to file input field");

            IWebElement uploadButton = driver.FindElement(By.Id("file-submit"));
            uploadButton.Click();
            Logger.Write("Clicked Upload button");

            // Validation
            string successHeader = driver.FindElement(By.TagName("h3")).Text;
            Logger.Write($"Upload confirmation header: '{successHeader}'");
            Assert.That(successHeader, Is.EqualTo("File Uploaded!"));

            string uploadedFileName = driver.FindElement(By.Id("uploaded-files")).Text;
            Logger.Write($"Uploaded file name: '{uploadedFileName}'");
            Assert.That(uploadedFileName, Is.EqualTo("sample_upload.txt"));

            // Cleanup
            if (File.Exists(tempFilePath))
            {
                File.Delete(tempFilePath);
                Logger.Write("Temporary file deleted");
            }

            Logger.Write("ENDING Test --- Completed successfully: File Upload ---");
        }
    }
}
